Make build failing on Apple silicon - macbook m2 pro

<!-- Thanks for helping us to improve LocalAI! We welcome all bug reports. Please fill out each area of the template so we can better help you. Comments like this will be hidden when you post but you can delete them if you wish. -->

**LocalAI version:**
<!-- Container Image or LocalAI tag/commit -->
728f297bb8c390d0de7b95b1969cbf1ad1864ab4

**Environment, CPU architecture, OS, and Version:**
<!-- Provide the output from "uname -a", HW specs, if it's a VM  -->
arm64(mac m2 pro) Ventura 13.3.1 

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
make build failing for mac m2 pro, 

```
Error 
LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:471:19: error: variable has incomplete type 'struct stat64'
```
**To Reproduce**
<!-- Steps to reproduce the behavior, including the LocalAI command used, if any -->
git clone the repo
`make build` on mac m2 pro

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
build should not fail

**Logs**
<!-- If applicable, add logs while running LocalAI in debug mode (`--debug` or `DEBUG=true`) to help explain your problem.  -->
```
make build

go mod edit -replace github.com/go-skynet/go-llama.cpp=/Users/ankur/dev/generative-ai/LocalAI/go-llama
go mod edit -replace github.com/nomic-ai/gpt4all/gpt4all-bindings/golang=/Users/ankur/dev/generative-ai/LocalAI/gpt4all/gpt4all-bindings/golang
go mod edit -replace github.com/go-skynet/go-ggml-transformers.cpp=/Users/ankur/dev/generative-ai/LocalAI/go-ggml-transformers
go mod edit -replace github.com/donomii/go-rwkv.cpp=/Users/ankur/dev/generative-ai/LocalAI/go-rwkv
go mod edit -replace github.com/ggerganov/whisper.cpp=/Users/ankur/dev/generative-ai/LocalAI/whisper.cpp
go mod edit -replace github.com/go-skynet/go-bert.cpp=/Users/ankur/dev/generative-ai/LocalAI/go-bert
go mod edit -replace github.com/go-skynet/bloomz.cpp=/Users/ankur/dev/generative-ai/LocalAI/bloomz
go mod edit -replace github.com/mudler/go-stable-diffusion=/Users/ankur/dev/generative-ai/LocalAI/go-stable-diffusion
go mod download
cd go-rwkv && cd rwkv.cpp &&	cmake . -DRWKV_BUILD_SHARED_LIBRARY=OFF &&	cmake --build . && 	cp librwkv.a .. && cp ggml/src/libggml.a ..
-- Accelerate framework found
-- CMAKE_SYSTEM_PROCESSOR: arm64
-- ARM detected
-- CMAKE_SYSTEM_PROCESSOR: arm64
-- ARM detected
-- Accelerate framework found
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp
[ 25%] Built target ggml
[ 37%] Building CXX object CMakeFiles/rwkv.dir/rwkv.cpp.o
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:209:5: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
    sprintf(full_key, "blocks.%d.%s", block_index, key.c_str());
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:471:19: error: variable has incomplete type 'struct stat64'
    struct stat64 file_stat;
                  ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:471:12: note: forward declaration of 'stat64'
    struct stat64 file_stat;
           ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:638:30: warning: cast from 'const rwkv_context *' to 'rwkv_context *' drops const qualifier [-Wcast-qual]
    ((struct rwkv_context *) ctx)->last_error = RWKV_ERROR_NONE;
                             ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:640:31: warning: cast from 'const rwkv_context *' to 'rwkv_context *' drops const qualifier [-Wcast-qual]
    RWKV_CTX_ASSERT_FALSE_MSG(ctx, RWKV_ERROR_ARGS, state_out != NULL, "state_out is NULL");
                              ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:641:31: warning: cast from 'const rwkv_context *' to 'rwkv_context *' drops const qualifier [-Wcast-qual]
    RWKV_CTX_ASSERT_FALSE_MSG(ctx, RWKV_ERROR_ARGS, logits_out != NULL, "logits_out is NULL");
                              ^
/Users/ankur/dev/generative-ai/LocalAI/go-rwkv/rwkv.cpp/rwkv.cpp:642:31: warning: cast from 'const rwkv_context *' to 'rwkv_context *' drops const qualifier [-Wcast-qual]
    RWKV_CTX_ASSERT_FALSE_MSG(ctx, RWKV_ERROR_ARGS, token < ctx->model->n_vocab, "Token is out of range 0..%d", ctx->model->n_vocab - 1);
                              ^
5 warnings and 1 error generated.
make[3]: *** [CMakeFiles/rwkv.dir/rwkv.cpp.o] Error 1
make[2]: *** [CMakeFiles/rwkv.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [go-rwkv/librwkv.a] Error 2
```


**Additional context**
<!-- Add any other context about the problem here. -->

--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,8 @@ GOLLAMA_VERSION?=10caf37d8b73386708b4373975b8917e6b212c0e
 GPT4ALL_REPO?=https://github.com/nomic-ai/gpt4all
 GPT4ALL_VERSION?=022f1cabe7dd2c911936b37510582f279069ba1e
 GOGGMLTRANSFORMERS_VERSION?=17b065584ef8f3280b6286235f0db95eec8a4b1c
-RWKV_REPO?=https://github.com/donomii/go-rwkv.cpp
-RWKV_VERSION?=ccb05c3e1c6efd098017d114dcb58ab3262b40b2
+RWKV_REPO?=https://github.com/mudler/go-rwkv.cpp
+RWKV_VERSION?=f25c89f8e55a67d57c01661a16abeed1b1c25016
 WHISPER_CPP_VERSION?=5b9e59bc07dd76320354f2af6be29f16dbcb21e7
 BERT_VERSION?=0548994371f7081e45fcf8d472f3941a12f179aa
 BLOOMZ_VERSION?=1834e77b83faafe912ad4092ccf7f77937349e2f
@@ -96,7 +96,7 @@ go-rwkv:
 	@find ./go-rwkv -type f -name "*.h" -exec sed -i'' -e 's/ggml_/ggml_rwkv_/g' {} +
 
 go-rwkv/librwkv.a: go-rwkv
-	cd go-rwkv && cd rwkv.cpp &&	cmake . -DRWKV_BUILD_SHARED_LIBRARY=OFF &&	cmake --build . && 	cp librwkv.a .. && cp ggml/src/libggml.a ..
+	cd go-rwkv && cd rwkv.cpp &&	cmake . -DRWKV_BUILD_SHARED_LIBRARY=OFF &&	cmake --build . && 	cp librwkv.a ..
 
 ## bloomz
 bloomz:

